File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevMap = {}
        for i in range(0, len(nums)):
            diff = target - nums[i]
            if prevMap.get(diff) is not None: return [prevMap[diff], i]
            else: prevMap[nums[i]] = i

File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val intMap = mutableMapOf<Int, Int>()
        nums.forEachIndexed { index, value -> 
            if (intMap.containsKey(target - value)) {
                return intArrayOf(intMap.getValue(target - value), index)
            }
            intMap[value] = index
        }
        throw IllegalArgumentException("No two sum solution")
    }
}